src/OpenApi/src/Extensions/ApiDescriptionExtensions.cs

```diff
@@ -70,4 +73,34 @@ public static string MapRelativePathToItemPath(this ApiDescription apiDescriptio
         }
         return strippedRoute.ToString();
     }
+
+    /// <summary>
+    /// Retrieves the form parameters from the ApiDescription, if they exist.
+    /// </summary>
+    /// <param name="apiDescription">The ApiDescription to resolve form parameters from.</param>
+    /// <param name="formParameters">A list of <see cref="ApiParameterDescription"/> associated with the form parameters.</param>
+    /// <returns><see langword="true"/> if form parameters were found, <see langword="false"/> otherwise.</returns>
+    public static bool TryGetFormParameters(this ApiDescription apiDescription, out IEnumerable<ApiParameterDescription> formParameters)
+    {
+        formParameters = apiDescription.ParameterDescriptions.Where(parameter => parameter.Source == BindingSource.Form || parameter.Source == BindingSource.FormFile);
+        return formParameters.Any();
+    }
+
+    /// <summary>
+    /// Retrieves the body parameter from the ApiDescription, if it exists.
+    /// </summary>
+    /// <param name="apiDescription">The ApiDescription to resolve the body parameter from.</param>
+    /// <param name="bodyParameter">The <see cref="ApiParameterDescription"/> associated with the body parameter.</param>
+    /// <returns><see langword="true"/> if a single body parameter was found, <see langword="false"/> otherwise.</returns>
+    public static bool TryGetBodyParameter(this ApiDescription apiDescription, [NotNullWhen(true)] out ApiParameterDescription? bodyParameter)
+    {
+        bodyParameter = null;
+        var bodyParameters = apiDescription.ParameterDescriptions.Where(parameter => parameter.Source == BindingSource.Body);
+        if (bodyParameters.Count() == 1)
+        {
+            bodyParameter = bodyParameters.Single();
```

Comment:
Use `SingleOrDefault()`?

Reply:
I think that throws if there are multiple elements?  (Unless you meant under the if, in which case, why?)

Reply:
I would have gone the other way and suggested `First` since there's no reason to do another bounds check.

Reply:
Oh that's me misremembering then. I thought Single() was one or exception, and SingleOrDefault() was one or null/default.

Reply:
`SingleOrDefault` will only return null/default if the list is empty. If there are multiple elements, it'll through an exception. Although based on the check that we are doing in the if, that should never be the case hence my inclination to use `Single`. `First` could also be used here but I feel it obfuscates the intention that there should always be only one parameter resolved from the JSON request body in a given request.